File: scala/sdktest/MoveTidEntityTest.scala
// Generated basic-flow test for the move_tid entity (model-driven;
// mirrors the java TestEntity generator). A dependency-free scala-cli test
// object driven by SdkEntityTestMain. Runs against the in-memory test
// transport seeded with the shipped MoveTidTestData.json fixtures.

import java.util.{ArrayList, LinkedHashMap, List => JList, Map => JMap}

import voxgig.bluefintecsmerchantservicessdk.core.{Helpers, BluefinTecsMerchantServicesSDK}
import voxgig.bluefintecsmerchantservicessdk.utility.struct.Struct

object MoveTidEntityTest {

  def run(rep: SdkTestReport): Unit = {
    rep.scope("move_tid.instance") {
      val testsdk = BluefinTecsMerchantServicesSDK.testSDK()
      val ent = testsdk.moveTid(null)
      rep.check("move_tid.instance", ent != null, "expected non-null move_tid entity")
    }

    rep.scope("move_tid.basic") {
      val entityData = Helpers.toMapAny(SdkTestSupport.readJson(
          "../.sdk/test/entity/move_tid/MoveTidTestData.json"))
      val options = new LinkedHashMap[String, Object]()
      options.put("entity", entityData.get("existing"))
      val client = BluefinTecsMerchantServicesSDK.testSDK(options, null)

      val idmap = new LinkedHashMap[String, Object]()
      idmap.put("move_tid01", "MOVE_TID01")
      idmap.put("move_tid02", "MOVE_TID02")
      idmap.put("move_tid03", "MOVE_TID03")
      val now = System.currentTimeMillis()

      // CREATE
      val moveTidRef01Ent = client.moveTid(null)
      var moveTidRef01Data = Helpers.toMapAny(Struct.getprop(
          Struct.getpath(entityData, "new.move_tid"), "move_tid_ref01"))
      val moveTidRef01DataResult = moveTidRef01Ent.create(moveTidRef01Data, null)
      moveTidRef01Data = Helpers.toMapAny(moveTidRef01DataResult match { case e: SdkEntity => e.data(); case o => o })
      rep.check("move_tid.create.map", moveTidRef01Data != null, "expected create result to be a map")
    }
  }
}

File: elixir/test/move_tid_entity_test.exs
# MoveTid entity test (offline, mock transport)

defmodule BluefinTecsMerchantServices.MoveTidEntityTest do
  use ExUnit.Case

  alias Voxgig.Struct, as: S
  alias BluefinTecsMerchantServices.Helpers, as: H
  alias BluefinTecsMerchantServices.Json

  defp fixture do
    Json.parse(File.read!("../.sdk/test/entity/move_tid/MoveTidTestData.json"))
  end

  defp mk_sdk do
    existing = H.or_(S.getpath(fixture(), "existing"), S.jm([]))
    BluefinTecsMerchantServices.test(S.jm(["entity", existing]))
  end

  defp first_id do
    existing = H.or_(S.getpath(fixture(), "existing.move_tid"), S.jm([]))
    keys = S.keysof(existing)
    if keys == [], do: nil, else: hd(keys)
  end

  test "should create instance" do
    sdk = BluefinTecsMerchantServices.test()
    ent = BluefinTecsMerchantServices.move_tid(sdk)
    assert ent != nil
  end

  test "should create then read back" do
    sdk = BluefinTecsMerchantServices.test(S.jm(["entity", S.jm(["move_tid", S.jm([])])]))
    ent = BluefinTecsMerchantServices.move_tid(sdk)
    created = BluefinTecsMerchantServices.Entity.MoveTid.create(ent, S.jm(["name", "test-create"]))
    made = BluefinTecsMerchantServices.EntityBase.data_get(created)
    assert S.ismap(made)
    assert S.getprop(made, "id") != nil
  end
end
